.pre-commit-config.yaml

```diff
@@ -38,18 +28,14 @@ repos:
         types: [ "jupyter" ]
         args: [ "--drop-empty-cells",
                 "--extra-keys 'metadata.language_info.version cell.metadata.jp-MarkdownHeadingCollapsed cell.metadata.pycharm'" ]
-  - repo: https://github.com/pycqa/flake8
-    rev: 4.0.1
-    hooks:
-      - id: flake8
-        types: ["python"]
-        additional_dependencies: ["flake8-bugbear==22.10.27"]
-  - repo: https://github.com/pycqa/bandit
-    rev: 1.7.4
-    hooks:
-      - id: bandit
-        additional_dependencies: ["bandit[toml]"]
-        args: ["-c", "pyproject.toml"]
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.3.7
+    hooks:
+      - id: ruff-format
+        types_or: [ python, pyi ]
+      - id: ruff
+        args: [ --fix, --exit-non-zero-on-fix ]
```

Comment:
I don't think `--exit-non-zero-on-fix` is really required here, pre-commit should catch any changed files.

Reply:
You are right. And the docs even recommend running the linter before the formatter. Changed it now